sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/async_challenge_auth_policy.py

```diff
@@ -55,3 +79,19 @@ class AsyncChallengeAuthPolicy(ChallengeAuthPolicyBase, AsyncHTTPPolicy):
 
         access_token = await self._credential.get_token(scope)
         self._update_headers(request.http_request.headers, access_token.token)
+
+    @staticmethod
+    def _log_request(request):
+        header_string = ""
+        for header in request.http_request.headers:
+            header_string += (", " + header)
+        if header_string == "":
+            _LOGGER.info("Sending {} request to url {} with no headers".format(
+                request.http_request.method, request.http_request.url
+            ))
+        else:
+            _LOGGER.info("Sending {} request to url {} with the following headers: {}".format(
+                request.http_request.method,
+                request.http_request.url,
+                header_string[2:]
```

Comment:
Why drop the first two characters?

Reply:
That was to get rid of fencepost characters, but solved by using the join method you suggested